daml-script/runner/src/test/scala/com/daml/lf/engine/script/TlsRunnerMainTest.scala

```diff
@@ -0,0 +1,211 @@
+  self: Suite =>
+
+  override protected lazy val tlsEnable: Boolean = true
+
+  private val tlsArgs: Seq[String] = Seq(
+    "--crt",
+    BazelRunfiles.rlocation("test-common/test-certificates/server.crt"),
+    "--pem",
+    BazelRunfiles.rlocation("test-common/test-certificates/server.pem"),
+    "--cacrt",
+    BazelRunfiles.rlocation("test-common/test-certificates/ca.crt"),
+    "--tls",
+  )
+
+  "TLS" - {
+    "GRPC" - {
+      "Succeeds with single run, no-upload" in
+        testDamlScriptCanton(
+          dars(0),
+          Seq(
+            "--ledger-host",
+            "localhost",
+            "--ledger-port",
+            ports.head.toString,
+            "--script-name",
+            "TestScript:myScript",
+          ) ++ tlsArgs,
+          Right(Seq("Ran myScript")),
+          Some(false),
+        )
+      // Checks we upload following the legacy behaviour, and throw our warning
+      "Succeeds with all run, no-upload-flag, default uploading behaviour" in
+        testDamlScriptCanton(
+          dars(1),
+          Seq(
+            "--ledger-host",
+            "localhost",
+            "--ledger-port",
+            ports.head.toString,
+            "--all",
+          ) ++ tlsArgs,
+          Right(
+            Seq(
+              "WARNING: Implicitly using the legacy behaviour",
+              "TestScript:myOtherScript SUCCESS",
+              "TestScript:myScript SUCCESS",
+            )
+          ),
+          Some(true),
+        )
+      "Succeeds with all run, explicit no-upload" in
+        testDamlScriptCanton(
+          dars(2),
+          Seq(
+            "--ledger-host",
+            "localhost",
+            "--ledger-port",
+            ports.head.toString,
+            "--all",
+            "--upload-dar=no",
+          ) ++ tlsArgs,
+          Right(
+            Seq(
+              "TestScript:myOtherScript SUCCESS",
+              "TestScript:myScript SUCCESS",
+            )
+          ),
+          Some(false),
+        )
+      "Succeeds with single run, explicit upload" in
+        testDamlScriptCanton(
+          dars(3),
+          Seq(
+            "--ledger-host",
+            "localhost",
+            "--ledger-port",
+            ports.head.toString,
+            "--script-name",
+            "TestScript:myScript",
+            "--upload-dar=yes",
+          ) ++ tlsArgs,
+          Right(Seq("Ran myScript")),
+          Some(true),
+        )
+      "Succeeds with single run, passing argument" in
+        testDamlScriptCanton(
+          dars(4),
+          Seq(
+            "--ledger-host",
+            "localhost",
+            "--ledger-port",
+            ports.head.toString,
+            "--script-name",
+            "TestScript:inputScript",
+            "--input-file",
+            inputFile,
+          ) ++ tlsArgs,
+          Right(Seq("Got 5")),
+        )
+      "Fails without TLS args" in
+        testDamlScriptCanton(
+          dars(4),
```

Comment:
```
// We use a different DAR for each test that asserts upload behaviour to avoid clashes
  val dars: Seq[Path] = (1 to DAR_COUNT).map(n =>
    BazelRunfiles.rlocation(Paths.get(s"daml-script/runner/test-script$n.dar"))
  )
```
We assert upload behaviour with that last argument to `testDamlScriptCanton`, so every call with a `Some(?)` at the end needs a different dar. The rest do not.